src/PowerShellEditorServices/Language/LanguageService.cs

```diff
@@ -294,72 +296,83 @@ public FindOccurrencesResult FindSymbolsInFile(ScriptFile scriptFile)
             ScriptFile[] referencedFiles,
             Workspace workspace)
         {
-            if (foundSymbol != null)
+            if (foundSymbol == null)
             {
-                int symbolOffset = referencedFiles[0].GetOffsetAtPosition(
-                    foundSymbol.ScriptRegion.StartLineNumber,
-                    foundSymbol.ScriptRegion.StartColumnNumber);
+                return null;
+            }
 
-                // Make sure aliases have been loaded
-                await GetAliases();
+            int symbolOffset = referencedFiles[0].GetOffsetAtPosition(
+                foundSymbol.ScriptRegion.StartLineNumber,
+                foundSymbol.ScriptRegion.StartColumnNumber);
 
-                // We want to look for references first in referenced files, hence we use ordered dictionary
-                var fileMap = new OrderedDictionary(StringComparer.OrdinalIgnoreCase);
-                foreach (ScriptFile file in referencedFiles)
-                {
-                    fileMap.Add(file.FilePath, file);
-                }
+            // Make sure aliases have been loaded
+            await GetAliases();
 
-                var allFiles = workspace.EnumeratePSFiles();
-                foreach (var file in allFiles)
+            // We want to look for references first in referenced files, hence we use ordered dictionary
+            // TODO: File system case-sensitivity is based on filesystem not OS, but OS is a much cheaper heuristic
+            var fileMap = RuntimeInformation.IsOSPlatform(OSPlatform.Linux)
+                ? new OrderedDictionary()
+                : new OrderedDictionary(StringComparer.OrdinalIgnoreCase);
+            foreach (ScriptFile file in referencedFiles)
+            {
+                fileMap.Add(file.FilePath, file);
+            }
+
+            foreach (string file in workspace.EnumeratePSFiles())
+            {
+                if (!fileMap.Contains(file))
                 {
-                    if (!fileMap.Contains(file))
+                    ScriptFile scriptFile;
+                    try
                     {
-                        fileMap.Add(file, new ScriptFile(file, null, this.powerShellContext.LocalPowerShellVersion.Version));
+                        scriptFile = workspace.GetFile(file);
```

Comment:
this is the change that will increase perf, yeah @SeeminglyScience?

Reply:
Yeah that constructor for `ScriptFile` (which is now unused, but is public, so...) does a call to `File.ReadAllText()` -- so it always reads off the file system. I'm always averse to something like touching the filesystem in a constructor (and don't really understand why it's done in this constructor, since there's another one that takes a full string), but the real problem is that it's likely we already have the script file contents loaded, and if we don't, this will cache them too